app/models/message.rb

```diff
@@ -9,19 +10,24 @@ class Message < ActiveRecord::Base
 
   validates :content, presence: true
   validates :groups, length: { minimum: 1, message: I18n.t('model.message.need_groups') }
-  validate :in_group, unless: :is_admin
-  attr_accessor :is_admin
+  validate :in_group, unless: :by_admin
 
+  scope :by_admin, -> { where(by_admin: true) }
   scope :by_latest, -> { order(created_at: :desc) }
   scope :by_user, ->(user) { where(user: user) }
   scope :for_user, ->(user) { joins(groups: :users).merge(User.where(id: user.id)) }
+  scope :for_admin, -> { includes([:user, :groups, message_comments: :user]).by_admin.by_latest }
 
   def with_group(user)
     if user.present?
       groups.merge(user.groups).any?
     end
   end
 
+  def group_names
+    groups.map(&:name).join(', ');
```

Comment:
This will load all the groups (all fields and such) and then do the mapping in ruby.

It is better to use the database as much as possible: `groups.pluck(:name)'.join(', ')`